tools/gen-postmortem-metadata.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+#
+# gen-postmortem-metadata.py output_file.cc
+#
+# Creates debugging symbols to help naviage Node's internals using post-mortem
+# debugging tools.
+#
+
+import sys
+
+
+class DebugSymbol(object):
+    type_ = 'int'
+    _PREFIX = 'nodedbg_'
+
+    def __init__(self, name, value, headers=[], type_=None):
+        self.name = name
+        self.value = value
+        self.headers = headers
+        self.type_ = type_ or DebugSymbol.type_
+
+    @classmethod
+    def get_headers(cls, debug_symbols):
+        seen = set()
+        seen_add = seen.add  # faster than seen.add on each iteration
+        headers = [debug_symbol.headers for debug_symbol in debug_symbols]
+        headers = sum(headers, [])
+
+        return [h for h in headers if not (h in seen or seen_add(h))]
+
+    def __str__(self):
+        return "{type} {prefix}{name} = {value};".format(
+            type=self.type_,
+            prefix=self._PREFIX,
+            name=self.name,
+            value=self.value,
+        )
+
+
+debug_symbols = [
+    DebugSymbol(
+        name="environment_context_idx_embedder_data",
+        value="Environment::kContextEmbedderDataIndex",
+        headers=["env.h"],
+        type_="int",
+    ),
+    DebugSymbol(
+        name="class__BaseObject__persistent_handle",
+        value="(size_t) offsetof(BaseObject, persistent_handle_)",
```

Comment:
Unnecessary cast here and elsewhere: the return type of `offsetof()` is already `size_t`.